Sources/SwiftFormat/PrettyPrint/Comment.swift

```diff
@@ -21,13 +21,27 @@ extension StringProtocol {
   /// - Returns: The string with trailing whitespace removed.
   func trimmingTrailingWhitespace() -> String {
     if isEmpty { return String() }
-    let scalars = unicodeScalars
-    var idx = scalars.index(before: scalars.endIndex)
-    while scalars[idx].properties.isWhitespace {
-      if idx == scalars.startIndex { return String() }
-      idx = scalars.index(before: idx)
+    let utf8Array = Array(utf8)
+    var idx = utf8Array.endIndex - 1
+    while utf8Array[idx].isWhitespace {
+      if idx == utf8Array.startIndex { return String() }
+      idx -= 1
+    }
+    return String(decoding: utf8Array[...idx], as: UTF8.self)
+  }
+}
```

Comment:
I have updated the formatter's whitespace criteria to match the linter's.
The formatter no longer considers characters like `U+2028` as whitespace, so they will no longer be removed.

This change has no significant impact on performance.
```
.build/release/swift-format lint --measure-instructions --recursive /tmp/swift-syntax

before: 50321785572
after: 49557836028
```